### PR TITLE
Switch to vite

### PR DESCRIPTION
**Pull request by dnlkoch**
This changes the bundler from WebPack to vite.

:exclamation: This includes BREAKING CHANGES as the docker setup has to be slightly adjusted.

1. `shogun-admin/config/admin-client-config.js`:

```diff
  path: {
-   modelConfigs: './modelconfigs',
+   modelConfigs: '/admin/modelconfigs',
    shogunBase: '/',
    logo: null
  },
}
```

2. `shogun-nginx/dev/default.conf`:

```diff
  location /admin/ {
-    proxy_pass http://shogun-admin:8080/;
+    proxy_pass http://shogun-admin:8080/admin/;

    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";

    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Port $server_port;
  }
```

### CODE REVIEW

**Comment by KaiVolland** [comment]
Please check #635 before merging.

File: vite.config.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      less: {
+        math: 'always',
+        relativeUrls: true,
+        javascriptEnabled: true,
+        modifyVars: theme
+      },
+    },
+  },
+  define: {
+    PROJECT_VERSION: JSON.stringify(process.env.npm_package_version),
```
Comment (dnlkoch) [inline vite.config.js:21]:
```suggestion
    PROJECT_VERSION: JSON.stringify(process.env.npm_package_version)
```

File: vite.config.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  plugins: [
+    react(),
+    viteExternalsPlugin({
+      shogunApplicationConfig: 'shogunApplicationConfig'
+    }),
```
Comment (dnlkoch) [inline vite.config.js:27]:
```suggestion
    })
```

File: vite.config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  ],
+  server: {
+    host: '0.0.0.0',
+    port: 8080,
+    strictPort: true,
```
Comment (dnlkoch) [inline vite.config.js:32]:
```suggestion
    strictPort: true
```

**Review by hwbllmnn — APPROVED** [review]

File: src/custom.d.ts [hunk 4]
```diff
@@ -199,3 +199,5 @@ declare type FormTranslations = {
     [localeKey: string]: string;
   };
 };
+
+declare module 'monaco-editor/esm/vs/*?worker'
```
Comment (dnlkoch) [inline src/custom.d.ts:203]:
```suggestion
declare module 'monaco-editor/esm/vs/*?worker';
```

**Comment by dnlkoch** [comment]
There is no breaking change commit available yet, do we want to mark this change as such?

**Comment by KaiVolland** [comment]
I will squash it and create a valid commit-message :crossed_fingers:

**Pull request merged**